### ISSUE TITLE
Use numerically stable sigmoid for PISING Glauber rates

### ISSUE THREAD

**Issue by haasonsaas** [extropic-ai/torx#15]
## Severity

**High for training; trivial fix** — forward values can look saturated/reasonable while backpropagation becomes NaN.

## Affected version

- PyPI `extro-torx==0.0.1`
- Commit `769d2f90abdfda14798fceb521143f4b99d370da`

## Summary

The Glauber rate is implemented as:

```python
# torx/psc/gates/_generator.py
1.0 / (1.0 + jnp.exp(beta * dE))
```

This should be:

```python
jax.nn.sigmoid(-beta * dE)
```

## Verified (float32 gradients w.r.t. dE)

| β·dE | old gradient | stable gradient |
|------|--------------|-----------------|
| 50 | `-0.0` | tiny nonzero |
| 80 | `-0.0` | tiny nonzero |
| 90 | **NaN** | `0` / tiny |
| 100 | **NaN** | `0` / tiny |

## Suggested fix

Replace the hand-written formula with `jax.nn.sigmoid(-beta * dE)` (or an equivalent stable form).

## Regression tests

Finite-gradient checks for energy products across roughly `[-100, 100]`, not only small values.

**Comment by lockwo** [comment]
true, fixed in https://github.com/extropic-ai/torx/pull/22
